src/forms/EditDocument/validate.js

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable no-useless-escape */
+import * as stringSimilarity from 'string-similarity';
+import {sortFilesBySimilarity} from '../../common/fileUtils';
+
+const MIN_SIMILARITY = 0.65;
+const DOC_NAME_REGEXS = [
+  /(new)|(final)|(modified)|(saved?)|(updated?)|(edit)|(\([0-9]\))|(\[[0-9]\])/, // black listed words
+  /\.[a-z]{2,}/, // file extensions
+  /[0-9]{1,2}[\.\/\-][0-9]{1,2}[\.\/\-][0-9]{2,4}|[0-9]{8}/, // dates
+];
+
+const replaceWithSpaces = str =>
+  str
+    .replace(/-|_/gi, ' ')
+    .trim()
+    .toLowerCase();
+
+const removeFileExt = str => str.replace(/\.(.*)/, '');
+
+const validate = ({file_name}, fileNode, studyFiles = []) => {
+  let errors = {
+    file_name: {
+      blacklisted: null,
+      existing_similarity: null,
+      upload_similarity: null,
+      file_ext: null,
+      dates: null,
+    },
+  };
+
+  let cleanFileName = replaceWithSpaces(removeFileExt(fileNode.name));
+  const DOC_NAME_INPUT = replaceWithSpaces(file_name);
+
+  const similarDocs = sortFilesBySimilarity(
+    {name: DOC_NAME_INPUT},
+    studyFiles,
+    0.33,
+  );
+
+  const uploadedFileSimilarity = stringSimilarity.compareTwoStrings(
+    DOC_NAME_INPUT,
+    cleanFileName,
+  );
+
+  // more than a quarter 75% of the inputed Document Title bigrams matches names of
```

Comment:
I'm not sure what this comment means, but the code seems to imply that this is true if there is one or more docs that have a Dice similarity of 0.33 or more (twice the size of the intersecting bigrams by the sum of the size of the sets of bigrams).

Reply:
yeah I'll admit I don't understand the Dice similarity 💯 and was my attempt to explain